### ISSUE TITLE
Some E3SM tests failed with Fortran runtime error: Array bound mismatch

### ISSUE THREAD

**Issue by dqwu** [E3SM-Project/E3SM#5632]
This issue is reproducible with latest E3SM master branch on a few machines including pm-cpu and anlgce.

When running e3sm_developer test suite on ANL GCE nodes, SMS_D_Ln5.ne4_oQU240.F2010 (and a few other tests) failed with the following error in e3sm.log:
```
...
[4] At line 122 of file /scratch/wuda/E3SM/components/eam/src/chemistry/mozart/rate_diags.F90
[4] Fortran runtime error: Array bound mismatch for dimension 3 of array 'rxt_rates_vmr' (31/7)
[4]
[4] Error termination. Backtrace:
[5] At line 122 of file /scratch/wuda/E3SM/components/eam/src/chemistry/mozart/rate_diags.F90
[5] Fortran runtime error: Array bound mismatch for dimension 3 of array 'rxt_rates_vmr' (31/7)
...
```

@ndkeen also saw 7 fails on pm-cpu that all have this error:
```
55: At line 122 of file /global/u2/e/e3smtest/jenkins/workspace/ACME_Perlmutter_next/E3SM/components/eam/src/chemistry/mozart/rate_diags.F90
55: Fortran runtime error: Array bound mismatch for dimension 3 of array 'rxt_rates_vmr' (31/7)
```

**Comment by dqwu** [comment]
It is also reproduced on e3sm_developer_master_gnu (mappy), see https://my.cdash.org/test/78604159
```
Error termination. Backtrace:
At line 122 of file /home/e3sm-jenkins/jenkins-ws/workspace/mappy_e3sm_master/E3SM/components/eam/src/chemistry/mozart/rate_diags.F90
Fortran runtime error: Array bound mismatch for dimension 3 of array 'rxt_rates_vmr' (31/7)
```

**Comment by rljacob** [comment]
Its a gnu thing.  @ndkeen also saw this.

**Comment by rljacob** [comment]
Attention @hsiangheleellnl @tangq

**Comment by rljacob** [comment]
I thought this was the 2D data array problem but its something else.

**Comment by hsiangheleellnl** [comment]
The problem has been solved #5626
